### PR TITLE
Codechange: rename sound ids to make more sense.

### PR DESCRIPTION
**Pull request by frosch123**
## Motivation / Problem

The enum items in SoundFx roughly reflected, what the sound effects sounded like, when someone heard them without context. 
Many sound effects were mislabeled, like "plane crashing" for the "toyland plane takeoff"-effect, or attributing "truck" sounds to "busses". Many sounds had arbitrary names like "another jet overhead".

## Description

This PR renames almost all items in SoundFx, and adds comments, how/where they are used in the source.
Similar used sounds get similar names:
* DEPARTURE_ _vehicle type_
* BREAKDOWN_ _vehicle type_
* TAKEOFF_ _aircraft type_
* CONSTRUCTION_ _type_
* _industry type_ _ _animation/effect_
* _climate zone_ _ _effect_

## Limitations

Poor NML people who copied the non-sensical names :cry: 

The separation into two commits is just for reviewing. This PR is intended to be squashed.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/sound_type.h [hunk 1]
```diff
@@ -36,79 +36,79 @@ struct SoundEntry {
...
+	SND_11_UNUSED,                         ///< 15 == 0x0F  unused (2)
+	SND_12_EXPLOSION,                      ///< 16 == 0x10  Destruction, crashes, disasters, ...
+	SND_13_TRAIN_COLLISION,                ///< 15 == 0x11  Train+train crash
+	SND_14_CASHTILL,                       ///< 18 == 0x12  Income from cargo delivery
+	SND_15_BEEP,                           ///< 19 == 0x13  GUI button click
+	SND_16_NEWS_TICKER,                    ///< 20 == 0x14  News ticker
+	SND_17_SKID_PLANE,                     ///< 21 == 0x15  Plane landing / touching ground
+	SND_18_TAKEOFF_HELICOPTER,             ///< 22 == 0x16  Takeoff: helicopter
+	SND_19_DEPARTURE_OLD_RV_1,             ///< 23 == 0x17  Station departure: truck and old bus (1) (non-toyland)
+	SND_1A_DEPARTURE_OLD_RV_2,             ///< 24 == 0x18  Station departure: truck and old bus (2) (random variation of SND_19_BUS_START_PULL_AWAY) (non-toyland)
```
Comment (glx22) [inline src/sound_type.h]: "SND_19_BUS_START_PULL_AWAY" is renamed

**Review by LordAro — COMMENTED** [review]
What's the value in keeping the numbers in the sound names? They already (mostly) don't match the actual enum value...

**Comment by frosch123** [comment]
In general: read the comment above the SoundFx enum. There is a historical off-by-two by CS.
In purpose of this PR: with preserving the IDs, it's super-easy to see that I did not mess up when replacing :)

**Review by LordAro — APPROVED** [review]
Sure.

**Pull request merged**